### PR TITLE
Cassandra native types tests

### PR DESCRIPTION
**Pull request by conorbros**
- changes to `Message` for parsing cassandra types (move to separate PR?)
- Tests for Cassandra native types (date, int, ascii, etc)
- Started tests for Cassandra collections: only have list at the moment and am not testing selects yet as the `ResultValue` needs to have stuff added for parsing collections.

### CODE REVIEW

**Comment by conorbros** [comment]
FYI: This is waiting on #406 and #433.

File: shotover-proxy/tests/cassandra_int_tests/mod.rs [hunk 1]
```diff
@@ -19,6 +19,26 @@ enum ResultValue {
...
     Int(i32),
     Boolean(bool),
+    Uuid(uuid::Uuid),
+    Ascii(String),
+    BigInt(i64),
+    Blob(Vec<u8>),
+    Decimal(Vec<u8>),
+    Double(Vec<u8>),
+    Duration(Vec<u8>),
+    Float(Vec<u8>),
```
Comment (rukai) [inline shotover-proxy/tests/cassandra_int_tests/mod.rs:29]: Should we store this as a float?
And there seems to be other similar cases too.

Comment (conorbros) [reply shotover-proxy/tests/cassandra_int_tests/mod.rs:29]: Yeah, I'll switch them over.

File: shotover-proxy/tests/cassandra_int_tests/basic_driver_tests.rs [hunk 2]
```diff
@@ -140,38 +140,431 @@ mod udt {
...
+                        "INSERT INTO test_collections_keyspace.test_list_table_{}_1 (id, my_list) VALUES ({}, [[{}]]);",
+                        i,
+                        i,
+                        get_type_example(*native_col_type)
+                    )
+                    .as_str(),
+                );
+            }
+
+            fn get_map_example(_key: usize, value: &str) -> String {
```
Comment (rukai) [inline shotover-proxy/tests/cassandra_int_tests/basic_driver_tests.rs:486]: unused arg?
Should we just inline this?

File: shotover-proxy/tests/cassandra_int_tests/basic_driver_tests.rs [hunk 3]
```diff
@@ -140,38 +140,431 @@ mod udt {
...
+                        "INSERT INTO test_collections_keyspace.test_list_table_{}_0 (id, my_list) VALUES ({}, [{{{}}}]);",
+                        i,
+                        i,
+                        get_type_example(*native_col_type)
+                    )
+                    .as_str(),
+                );
+            }
+
+            // // test inserting list of lists
```
Comment (rukai) [inline shotover-proxy/tests/cassandra_int_tests/basic_driver_tests.rs:472]: double comment

File: shotover-proxy/tests/cassandra_int_tests/basic_driver_tests.rs [hunk 4]
```diff
@@ -140,38 +140,431 @@ mod udt {
...
+        }
+
+        fn select(session: &Session) {
+            for (i, col_type) in NATIVE_COL_TYPES.iter().enumerate() {
+                let query = format!(
+                    "SELECT my_list FROM test_collections_keyspace.test_list_table_{};",
+                    i
+                );
+
+                // , i, i, get_type_example(*col_type));
```
Comment (rukai) [inline shotover-proxy/tests/cassandra_int_tests/basic_driver_tests.rs:512]: commented code?

File: shotover-proxy/tests/cassandra_int_tests/basic_driver_tests.rs [hunk 5]
```diff
@@ -140,38 +140,431 @@ mod udt {
...
+            ColType::Inet => ResultValue::Inet("127.0.0.1".into()),
+            ColType::Date => ResultValue::Date(vec![128, 0, 58, 160]),
+            ColType::Time => ResultValue::Time(vec![0, 0, 26, 229, 187, 195, 188, 0]),
+            ColType::Smallint => ResultValue::SmallInt(32767),
+            ColType::Tinyint => ResultValue::TinyInt(127),
+            _ => panic!("dont have an example for {}", col_type),
+        }
+    }
+
+    mod list {
```
Comment (rukai) [inline shotover-proxy/tests/cassandra_int_tests/basic_driver_tests.rs:415]: this list module seems weird, not everything in here is just for lists...

Comment (conorbros) [reply shotover-proxy/tests/cassandra_int_tests/basic_driver_tests.rs:415]: It's inserting a list of other collections, so technically it is testing lists. I have other work waiting to go in to test maps and sets in the same way.

Comment (rukai) [reply shotover-proxy/tests/cassandra_int_tests/basic_driver_tests.rs:415]: nah, if theres context im missing from future PRs thats fine, leave it as is.
We can always refactor it later if we decide it looks weird after everything is in.

**Review by rukai — APPROVED** [review]
Nice! Very thorough coverage so far

**Review by benbromhead — APPROVED** [review]
Nice!!! Good set of tests that will keep us on the straight and narrow :dancers:

**Pull request merged**